pkgs/development/compilers/oraclejdk/jdk-linux-base.nix

```diff
@@ -36,24 +37,33 @@
 
 assert swingSupport -> xorg != null;
 
+if !licenseAccepted then throw ''
+    You must accept the Oracle Binary Code License Agreement for Java SE at
+    https://www.oracle.com/technetwork/java/javase/terms/license/index.html
+    by adding `oraclejdk.accept_license = true;` to your `.nix/config.nix`.
+  ''
+else assert licenseAccepted;
+
 let
 
   /**
    * The JRE libraries are in directories that depend on the CPU.
    */
-  architecture = {
+  architectures = {
     i686-linux    = "i386";
     x86_64-linux  = "amd64";
     armv7l-linux  = "arm";
     aarch64-linux = "aarch64";
-  }.${stdenv.hostPlatform.system};
+  };
+
+  architecture = architectures.${stdenv.hostPlatform.system};
 
   jce =
     if installjce then
-      requireFile {
-        name = jceName;
-        url = jceDownloadUrl;
+      fetchurl {
+        url = "http://download.oracle.com/otn-pub/java/jce/${productVersion}/${jceName}";
         sha256 = sha256JCE;
+        curlOpts = "-b oraclelicense=a"; # denote accepted license by setting magic cookie
```

Comment:
Does it work with only ```oraclelicense=a``` instead of usual ```oraclelicense=accept-securebackup-cookie``` ?
Or everyone suppose to patch its own nixpkgs copy to accept the license by doing so?

Reply:
I copied that from the https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=jdk and it works for me. So I expect this to work for everyone.